mesh/tests/chat_wsclient_pytests.py

```diff
@@ -0,0 +1,109 @@
+"""
+To run this test:
+
+pytest ./mesh/tests/message_client_tests.py
+
+# Inside Docker container:
+(Without activating `pipenv shell`)
+
+pipenv run pytest ./mesh/tests/message_client_tests.py
+"""
+
+import pytest
+import json
+from channels.testing import WebsocketCommunicator
+from mesh.conversation.models import Conversation, Message
+from mesh.accounts.models import Account
+from asgiref.sync import sync_to_async
+from django.test import Client
+from mesh.conversation.consumers import ChatConsumer
+from channels.routing import URLRouter
+from django.urls import path
+from django.conf import settings
+
+@pytest.mark.django_db
+@pytest.mark.asyncio
+async def test_chat_consumer():
+    """
+    Test the ChatConsumer for WebSocket communication between two clients.
+    The test covers connection, message sending, and message receiving.
+    """
+    # Use in-memory channel layers for testing.
+    settings.CHANNEL_LAYERS = {
+        'default': {
+            'BACKEND': 'channels.layers.InMemoryChannelLayer',
+        },
+    }
+    
+    client = Client()
+    
+    # Create a test conversation asynchronously
+    conversation = await sync_to_async(Conversation.objects.create)(conversationType=0)
+    conversationID = conversation.conversationID
+    
+    # Create test accounts asynchronously
+    account1 = await sync_to_async(Account.objects.create)(email='user1@example.com')
+    account2 = await sync_to_async(Account.objects.create)(email='user2@example.com')
+
+    # Add accounts to the conversation as participants
+    response = client.post(
+        '/conversations/', 
+        json.dumps({
```

Comment:
I believe the json.dumps method is redundant because of the content_type, not entirely sure